src/main/java/address/sync/CloudSimulator.java

```diff
@@ -1,129 +1,670 @@
+        LocalDateTime localDateTime = LocalDateTime.now();
+        ZonedDateTime zonedDateTime = localDateTime.atZone(ZoneOffset.systemDefault());
+        return zonedDateTime.getOffset();
+    }
+
+    private boolean shouldSimulateNetworkFailure() {
+        return shouldSimulateUnreliableNetwork && RANDOM_GENERATOR.nextDouble() <= FAILURE_PROBABILITY;
+    }
+
+    private boolean shouldSimulateSlowResponse() {
+        return shouldSimulateUnreliableNetwork && RANDOM_GENERATOR.nextDouble() <= FAILURE_PROBABILITY;
+    }
+
+    private RawCloudResponse getNetworkFailedResponse() {
+        System.out.println("Cloud simulator: failure occurred! Could not retrieve data");
+        return new RawCloudResponse(HttpURLConnection.HTTP_INTERNAL_ERROR, null, null);
+    }
+
+    private boolean isWithinQuota(int quotaUsed) {
+        System.out.println("Current quota left: " + rateLimitStatus.getQuotaRemaining() + ", using " + quotaUsed);
+        return quotaUsed <= rateLimitStatus.getQuotaRemaining();
+    }
+
+    private boolean isExistingPerson(List<CloudPerson> personList, CloudPerson targetPerson) {
+        return personList.stream()
+                .filter(person -> person.getFirstName().equals(targetPerson.getFirstName())
+                        && person.getLastName().equals(targetPerson.getLastName()))
+                .findAny()
+                .isPresent();
+    }
+
+    private boolean isExistingTag(List<CloudTag> tagList, CloudTag targetTag) {
+        return tagList.stream()
+                .filter(tag -> !targetTag.getName().equals(targetTag.getName()))
+                .findAny()
+                .isPresent();
+    }
+
+    private HashMap<String, String> getStandardHeaders() {
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put("X-RateLimit-Limit", String.valueOf(rateLimitStatus.getQuotaLimit()));
+        headers.put("X-RateLimit-Remaining", String.valueOf(rateLimitStatus.getQuotaRemaining()));
+        headers.put("X-RateLimit-Reset", String.valueOf(rateLimitStatus.getQuotaReset()));
+        return headers;
+    }
+
+    /**
+     * Verifies whether newPerson can be added, and adds it to the persons list
+     *
+     * @param personList
+     * @param newPerson
+     * @return newPerson, if added, else null
+     */
+    private CloudPerson addPerson(List<CloudPerson> personList, CloudPerson newPerson) throws JAXBException {
+        if (newPerson == null) throw new IllegalArgumentException("Person cannot be null");
+        String newPersonFirstName = newPerson.getFirstName();
+        String newPersonLastName = newPerson.getLastName();
+        if (newPersonFirstName == null || newPersonLastName == null) {
+            throw new IllegalArgumentException("Fields cannot be null");
+        }
+        if (isExistingPerson(personList, newPerson)) throw new IllegalArgumentException("Person already exists");
+
+        personList.add(newPerson);
+
+        return newPerson;
```

Comment:
I think this method is outdated. The comment says `return null if person not added` but I don't see it in the implementation. I also don't see anything that would throw `JAXBException` and instead it throws many `IllegalArgumentException`.